cask.el

```diff
@@ -833,17 +833,29 @@ URL is the url to the mirror."
                            (cask-bundle-sources bundle))))
     (setf (cask-bundle-sources bundle) sources)))
 
-(defun cask-build (bundle)
-  "Build BUNDLE Elisp files."
+(defun cask-build (bundle &rest args)
+  "Build BUNDLE Elisp files.
+
+Return t if all files were successfully built without errors,
+otherwise return nil.
```

Comment:
If we want Cask to exit with failure on compile errors we should use `error` to really abort compilation and not just return nil/non-nil.  IOW, we should not propagate exceptional conditions with return values.

Please refactor the code to use `error`s for compile errors.  If need be define a custom error signal (which I think we already do for Cask's own errors).